### ISSUE TITLE
"Oops something went wrong" with invalid username - Teacher Registration

### ISSUE THREAD

**Issue by jwzimmer** [LLK/scratch-www#1097]
Production, Chrome 54, Mac OS El Capitan

**TL;DR**
- Error visible to user on the username/ password step of registration
- Error in console
- No known impact on functionality; I completed registration after triggering this error

I'm seeing "Oops something went wrong" & 
"https://scratch.mit.edu/accounts/check_username/[username]/ Failed to load resource: the server responded with a status of 404 (NOT FOUND)" when clicking outside the username field when it has a current error (for invalid chars included in username).

To repro, enter a username like "mr.jwzimmer1990", then click outside the fields. You should see "Oops something went wrong" appear in the UI and a 404 in the console on the username check. 

<img width="1277" alt="screen shot 2016-12-05 at 5 25 09 pm" src="https://cloud.githubusercontent.com/assets/1320456/20905041/34b0317c-bb10-11e6-9098-78f75253eb02.png">

I'm concerned that in some cases this issue could prevent teachers from being able to sign up, if for some reason the error early in the process causes the whole process to fail (I'm trying to repro that).

**Comment by jwzimmer** [comment]
I was able to complete registration (got to waiting room) for a teacher account after triggering those errors on Production.
